spec/API_specification/array_api/array_object.py

```diff
@@ -706,16 +708,42 @@ def __mul__(self: array, other: Union[int, float, array], /) -> array:
         -   If ``x1_i`` is a nonzero finite number and ``x2_i`` is either ``+infinity`` or ``-infinity``, the result is a signed infinity with the mathematical sign determined by the rule already stated above.
         -   In the remaining cases, where neither ``infinity`` nor `NaN` is involved, the product must be computed and rounded to the nearest representable value according to IEEE 754-2019 and a supported rounding mode. If the magnitude is too large to represent, the result is an ``infinity`` of appropriate mathematical sign. If the magnitude is too small to represent, the result is a zero of appropriate mathematical sign.
 
+        For complex floating-point operands, multiplication is defined according to the following table. For real components ``a`` and ``c`` and imaginary components ``b`` and ``d``,
+
+        +------------+----------------+-----------------+--------------------------+
+        |            | c              | dj              | c + dj                   |
+        +============+================+=================+==========================+
+        | **a**      | a * c          | (a*d)j          | (a*c) + (a*d)j           |
+        +------------+----------------+-----------------+--------------------------+
+        | **bj**     | (b*c)j         | -(b*d)          | -(b*d) + (b*c)j          |
+        +------------+----------------+-----------------+--------------------------+
+        | **a + bj** | (a*c) + (b*c)j | -(b*d) + (a*d)j | implementation-dependent |
```

Comment:
The ` implementation-dependent` here is confusing, and inconsistent with the PR description. Why not add `(ac - bd) + (bc + ad)j` here? The "if a component is nan/inf" is already handled by special-case text below, now it looks like the whole operation isn't well-defined.

Reply:
The same comment applies to division (gh-554)

Reply:
I updated the entry from `implementation-dependent` to `special rules`. The main purpose of the table is to describe when the normal rules for real-valued floating-point case special cases apply. As the normal rules don't apply for multiplication of two complex numbers, IMO adding the textbook formula for cmplx-cmplx would also be misleading given the content which immediately follows the table.

My concern is that if the table shows only the textbook formula without indication that cmplx-cmplx may vary between implementations, readers won't continue reading on to find out that special case handling doesn't always apply.

Reply:
Wrt complex division, I should note that the guidance following the table is that, for finite components, division "should" (not "must") be computed as if by the textbook formula, as there are algorithms, similar to `hypot`, for avoiding underflow/overflow. And for complex multiplication, one can use [Karatsuba's algorithm](https://en.wikipedia.org/wiki/Karatsuba%27s_algorithm) to reduce the number of multiplications. In short, I am hesitant to be too prescriptive in what is allowed/disallowed.

Reply:
"special rules" seems better, thanks. After that section is moved to notes, so it's clearer that the whole thing is special case handling related, it will be even better. Let's roll with this